### PR TITLE
Replace our package json version fetch from graphql to octokit

### PR DESCRIPTION
**Pull request by gaber85**
This PR takes our package json version compare function currently relying on a deprecated Github GraphQL package and replaces it with Github's Octokit request endpoint.

- [x] Update tests to work with Octokit.

### CODE REVIEW

**Comment by gaber85** [comment]
Hey team, I've finally got around to resolving this failing action. I've replaced GraphQL with Octokit and updated the tests around it. Just tagging you so you can see what I did. @hamalamb @Billyhaine @RichardHpa

File: src/version.ts [hunk 1]
```diff
@@ -37,38 +37,24 @@ export function setPackageJsonVersion(config: Config = getConfig()): void {
 
 async function fetchLatestVersion(
   config: Config = getConfig()
-): Promise<GraphQlQueryResponseData> {
+) {
```
Comment (hamalamb) [inline src/version.ts:40]: Wondering if this is `Promise<OctokitResponse>`

from https://octokit.github.io/types.ts/types/OctokitResponse.html
looking through [@octokit/types](https://octokit.github.io/types.ts/)

Comment (gaber85) [reply src/version.ts:40]: Yes, that's the correct type for this function. Thankfully octokit is fully typed and this function correctly infers the return type here 👍

**Review by Billyhaine — APPROVED** [review]
Yeah nice LGTM, seems oktokit just had what we needed afterall

File: src/version.ts [hunk 2]
```diff
@@ -37,38 +37,24 @@ export function setPackageJsonVersion(config: Config = getConfig()): void {
...
-      `,
-      {
-        owner,
-        repo,
-        headers: {
-          authorization,
-        },
-      }
-    );
+    return await octokit.request('GET /repos/{owner}/{repo}/releases/latest', {
```
Comment (fsanggang) [inline src/version.ts:53]: 👍🏻 Yip, matches [docs](https://docs.github.com/en/rest/releases/releases#get-the-latest-release).

File: src/version.ts [hunk 3]
```diff
@@ -88,19 +74,13 @@ async function fetchLatestVersion(
...
   config: Config = getConfig()
 ): Promise<string> {
-  const response: GraphQlQueryResponseData = await fetchLatestVersion(config);
-
-  if (response.repository.packages.nodes.length > 1) {
-    throw new Error(
-      `🚧 Found ${response.repository.packages.nodes.length} packages on '${config.repository}', expected only 1.  Monorepo/multiple packages is not supported yet.`
-    );
-  }
+  const { data } = await fetchLatestVersion(config);
```
Comment (fsanggang) [inline src/version.ts:77]: 👍🏻 Yip, matches [examples](https://github.com/octokit/core.js#authentication), we get a `data` object back.

**Review by fsanggang — APPROVED** [review]
Great job with Octokit, and extra credit for the tests 😉  Let's see it in action with today's release _woop woop_

**Pull request merged**